### PR TITLE
Add new API version 2024-09-01 for Microsoft.Cdn

### PR DESCRIPTION
**Pull request by gubalasu**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [X] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [X] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [X] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Review by ChenglongLiu — APPROVED** [review]

**Comment by mentat9** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cdn/profiles/{profileName}/cdnCanMigrateToAfd": {

Action names should follow a <verb><object> pattern. In this case, maybe something like checkMigratability or validateReadyForMigration, or something similar.

---
Refers to: specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-09-01/cdn.json:2732 in 73e51e4. [](commit_id = 73e51e4da8eebd08d1d554ec460b6bde7717ee9b, deletion_comment = False)

**Comment by mentat9** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cdn/profiles/{profileName}/cdnMigrateToAfd": {

No need to include cdn in the name, since it's an action on a cdn profile. You could just use migrateToAfd, or startMigration, or startMigrationToAfd, etc.

---
Refers to: specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-09-01/cdn.json:2786 in 73e51e4. [](commit_id = 73e51e4da8eebd08d1d554ec460b6bde7717ee9b, deletion_comment = False)

**Comment by mentat9** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cdn/profiles/{profileName}/migrationAbort": {

<verb><object> => abortMigration, cancelMigration, stopMigration, etc. The best name for this might depend on the state that's left when someone executes this action. Does this action reset everything back to the original unmigrated state, or does it leave things in a partially migrated state, and another migration action is required to complete the migration? If it's the former, I think cancel is a better verb to use. If it's the latter, then stop might be better.

---
Refers to: specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-09-01/cdn.json:2849 in 73e51e4. [](commit_id = 73e51e4da8eebd08d1d554ec460b6bde7717ee9b, deletion_comment = False)

**Comment by mentat9** [comment]
>       "properties": {

If these two are URLs, consider adding format: uri and including that in the description.

---
Refers to: specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-09-01/cdn.json:7422 in 73e51e4. [](commit_id = 73e51e4da8eebd08d1d554ec460b6bde7717ee9b, deletion_comment = False)

**Comment by mentat9** [comment]
>           "description": "The name of the old endpoint.",

I find this description a little unexpected/confusing. Do these endpoints all have names? See if you can clarify how these are specified.

---
Refers to: specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-09-01/cdn.json:7424 in 73e51e4. [](commit_id = 73e51e4da8eebd08d1d554ec460b6bde7717ee9b, deletion_comment = False)

**Comment by mentat9** [comment]
>       "properties": {}

If the response doesn't have a body, use status code 204 with no schema rather than creating an empty property bag.

---
Refers to: specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-09-01/cdn.json:7436 in 73e51e4. [](commit_id = 73e51e4da8eebd08d1d554ec460b6bde7717ee9b, deletion_comment = False)

**Comment by gubalasu** [comment]
> > ```
> >   "properties": {
> > ```
> 
> If these two are URLs, consider adding format: uri and including that in the description.
> 
> Refers to: specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-09-01/cdn.json:7422 in [73e51e4](https://github.com/Azure/azure-rest-api-specs/commit/73e51e4da8eebd08d1d554ec460b6bde7717ee9b). [](commit_id = [73e51e4](https://github.com/Azure/azure-rest-api-specs/commit/73e51e4da8eebd08d1d554ec460b6bde7717ee9b), deletion_comment = False)

These are both names. I have updated the description. It should be clearer now.

**Comment by gubalasu** [comment]
> > ```
> >       "description": "The name of the old endpoint.",
> > ```
> 
> I find this description a little unexpected/confusing. Do these endpoints all have names? See if you can clarify how these are specified.
> 
> Refers to: specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-09-01/cdn.json:7424 in [73e51e4](https://github.com/Azure/azure-rest-api-specs/commit/73e51e4da8eebd08d1d554ec460b6bde7717ee9b). [](commit_id = [73e51e4](https://github.com/Azure/azure-rest-api-specs/commit/73e51e4da8eebd08d1d554ec460b6bde7717ee9b), deletion_comment = False)

Updated the description. It should be clearer now

**Comment by gubalasu** [comment]
> > ```
> > "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cdn/profiles/{profileName}/cdnMigrateToAfd": {
> > ```
> 
> No need to include cdn in the name, since it's an action on a cdn profile. You could just use migrateToAfd, or startMigration, or startMigrationToAfd, etc.
> 
> Refers to: specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-09-01/cdn.json:2786 in [73e51e4](https://github.com/Azure/azure-rest-api-specs/commit/73e51e4da8eebd08d1d554ec460b6bde7717ee9b). [](commit_id = [73e51e4](https://github.com/Azure/azure-rest-api-specs/commit/73e51e4da8eebd08d1d554ec460b6bde7717ee9b), deletion_comment = False)

We would ideally want to keep the names of all the actions as is because we already have customers using this in the preview version as it was already approved.
https://github.com/Azure/azure-rest-api-specs/pull/29369
These API action names are also similar to a related API that is used for a slightly different purpose but a similar functionality as this. So, keeping the names as is would help our customers understand the use of each of these APIs 
https://github.com/Azure/azure-rest-api-specs/blob/260651b67269ee08d515b2bae9d2e9f9c5732c42/specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-02-01/cdn.json#L413

@mentat9  This comment is applicale for all 3 comments related to the action names

**Comment by gubalasu** [comment]
@mentat9 Addressed and responded to your comments. Ready to take another look

**Comment by gubalasu** [comment]
@mentat9 Could you please help take another look at this?

**Comment by mentat9** [comment]
> @mentat9 Could you please help take another look at this?

@gubalasu - Please always carefully read and follow instructions in the Next Steps to Merge comment https://github.com/Azure/azure-rest-api-specs/pull/30309#issuecomment-2307451108 which is continually updated through the handling of your PR. Your PR is currently invisible to reviewers because it has the ARMChangesRequested label.

**Comment by gary-x-li** [comment]
Please fix the following failed checks:

Swagger LintDiff: https://github.com/Azure/azure-rest-api-specs/pull/30309/checks?check_run_id=29738908403
Swagger ModelValidation: https://github.com/Azure/azure-rest-api-specs/pull/30309/checks?check_run_id=29738909226

**Comment by gubalasu** [comment]
> Please fix the following failed checks:
> 
> Swagger LintDiff: https://github.com/Azure/azure-rest-api-specs/pull/30309/checks?check_run_id=29738908403 Swagger ModelValidation: https://github.com/Azure/azure-rest-api-specs/pull/30309/checks?check_run_id=29738909226

I have fixed the LintDiff. The ModelValidation is from the copy I did from older version and not from my PR. Can you take a look at it now.

**Comment by gubalasu** [comment]
> I understand the ModelValidation violations are not related to this PR. However, please fix them since they are bugs in your specs and it will make future PR authoring and reviewing process smoother.
> 
> Both violations are likely related to the absence of `id` property in `DeepCreatedOrigin`.

@gary-x-li I have fixed some of the other validation errors other than the DeepCreatedOrigin and DeepCreatedOriginGroup.
These is on deprecation and the changes in the swagger need equivalent changes on our RP and we are not doing changes on these resources anymore. Could you please check if everything else looks alright

**Comment by gary-x-li** [comment]
>           "final-state-via": "azure-async-operation"

Change to "location" and make sure that's the behavior of your service. It should be consistent with the header defined in 202 response above.

---
Refers to: specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-09-01/cdn.json:2893 in e18c234. [](commit_id = e18c234159f34b854f3c2ea2e3af647a0ef961aa, deletion_comment = False)

**Comment by gary-x-li** [comment]
> > I understand the ModelValidation violations are not related to this PR. However, please fix them since they are bugs in your specs and it will make future PR authoring and reviewing process smoother.
> > Both violations are likely related to the absence of `id` property in `DeepCreatedOrigin`.
> 
> @gary-x-li I have fixed some of the other validation errors other than the DeepCreatedOrigin and DeepCreatedOriginGroup. These is on deprecation and the changes in the swagger need equivalent changes on our RP and we are not doing changes on these resources anymore. Could you please check if everything else looks alright

Sounds good. I wonder whether it makes sense to remove `x-ms-azure-resource: true` from these definitions. https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/Semantic-and-Model-Violations-Reference.md#missing_resource_id

**Comment by gubalasu** [comment]
> > ```
> >       "final-state-via": "azure-async-operation"
> > ```
> 
> Change to "location" and make sure that's the behavior of your service. It should be consistent with the header defined in 202 response above.
> 
> Refers to: specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-09-01/cdn.json:2893 in [e18c234](https://github.com/Azure/azure-rest-api-specs/commit/e18c234159f34b854f3c2ea2e3af647a0ef961aa). [](commit_id = [e18c234](https://github.com/Azure/azure-rest-api-specs/commit/e18c234159f34b854f3c2ea2e3af647a0ef961aa), deletion_comment = False)

You are right. I overlooked it. Updated it now

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-09-01/cdn.json [hunk 1]
```diff
@@ -2728,6 +2728,171 @@
...
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cdn/profiles/{profileName}/cdnCanMigrateToAfd": {
+      "post": {
+        "tags": [
+          "Profiles"
+        ],
+        "description": "Checks if CDN profile can be migrated to Azure Frontdoor(Standard/Premium) profile.",
+        "operationId": "CdnProfiles_CdnCanMigrateToAfd",
```
Comment (tadelesh) [inline specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-09-01/cdn.json:2738]:
```suggestion
        "operationId": "Profiles_CdnCanMigrateToAfd",
```

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-09-01/cdn.json [hunk 2]
```diff
@@ -2728,6 +2728,171 @@
...
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cdn/profiles/{profileName}/cdnMigrateToAfd": {
+      "post": {
+        "tags": [
+          "Profiles"
+        ],
+        "description": "Migrate the CDN profile to Azure Frontdoor(Standard/Premium) profile. This step prepares the profile for migration and will be followed by Commit to finalize the migration.",
+        "operationId": "CdnProfiles_CdnMigrateToAfd",
```
Comment (tadelesh) [inline specification/cdn/resource-manager/Microsoft.Cdn/stable/2024-09-01/cdn.json:2792]:
```suggestion
        "operationId": "Profiles_CdnMigrateToAfd",
```

**Comment by gubalasu** [comment]
ARM team: I added a suppression as this is needed to bypass the ModelValidation. The errors were explained in the comments and @gary-x-li already approved it. Suppressions need to be approved to merge the PR

**Comment by ms-bogdan-rotaru** [comment]
Your branch is out-of-date with base branch, please merge with base branch before requesting merge.

**Review by rrahulms — APPROVED** [review]

**Pull request merged**